### PR TITLE
Add CUDA 11.8 cross-compilation support

### PR DESCRIPTION
**Pull request by jakirkham**
This adds additional logic to the CUDA 11 cross-compilation case here to support CUDA 11.8.

Needed as part of the CUDA 11.8 migration: https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/4834

<hr>

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Review by h-vetinari — COMMENTED** [review]
It would be nice if this actually turns out this painless (given the struggle to get this right for 11.2)!

I would still suggest to actually let this run through execution here before merging. At the time of #210, I had locally pulled in a (cross-enabled) 11.2 migrator, so that the CI would generate the respective jobs here and actually execute the branches in question.

In your case, this would be a cross-enabled version of https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/4834 (plus `local: True`).

I don't necessarily expect that nvidia's artefact naming has changed a lot between 11.2 & 11.8, but IMO better to double-check. As one concrete issue, the `extra="11-2-"` further down will need adaptation as well.

**Comment by jakirkham** [comment]
Thanks Axel! 🙏

Appreciate the helpful tips. Have tried to push some changes to address those

Please let me know how that looks

File: .ci_support/migrations/cuda118.yaml [hunk 1]
```diff
@@ -2,6 +2,7 @@ migrator_ts: 1692828152
 __migrator:
   kind:
     version
+  use_local: true # TODO: Remove this line after completing testing
```
Comment (h-vetinari) [inline .ci_support/migrations/cuda118.yaml:5]: Not sure that we want to have the cross-compiled jobs for merging. We don't _need_ them from the POV of the CI setup itself, just for testing the branches here in CI. I suggested adding them in https://github.com/conda-forge/conda-forge-ci-setup-feedstock/pull/236, but never got a response.

Comment (jakirkham) [reply .ci_support/migrations/cuda118.yaml:5]: Yep no strong feelings here. Happy to keep or drop as preferred

**Comment by jakirkham** [comment]
CI is failing as the `nvcc` wrapper script package has not been built for CUDA 11.8 cross-compilation

Addressing with PR: https://github.com/conda-forge/nvcc-feedstock/pull/99

**Comment by h-vetinari** [comment]
Tried to merge main here after #99, but it seem you've unchecked the box that says "allow edits by maintainers"

**Comment by jakirkham** [comment]
Thanks Axel! 🙏

Unfortunately since this is an org fork (as opposed to a user fork) that check box isn't even available (I couldn't check it if I wanted to)

Have clicked the update branch button. So we should be good to go

Also FWIW closing and reopening PRs has the same effect (it deletes the merge ref for the PR and recreates it using the latest upstream branch), which will also update CI as a result

**Comment by h-vetinari** [comment]
Ah, I had forgotten that this depends on https://github.com/conda-forge/nvcc-feedstock/pull/99. 🙃

**Comment by jakirkham** [comment]
Yeah no worries. This wound up being a bit of a yak shave (though that was mostly expected)

**Comment by h-vetinari** [comment]
I did this in #267 and it's passing! 🥳 

If we do not want to keep the cross-compilation jobs (which would be purely for testing purposes), then you could rebase/revert out the migrator and merge this from my POV. 👍

**Comment by jakirkham** [comment]
Thanks Axel! 🙏

Went ahead and cherry-picked the last 2 commits. Hope that is ok

**Comment by jakirkham** [comment]
Forgot to bump version 🤦‍♂️

Fixed with commit: https://github.com/conda-forge/conda-forge-ci-setup-feedstock/commit/22b4448b3d9feda2f78f94149b68a015366c3a9e

**Pull request merged**